query/src/sql/planner/binder/scalar.rs

```diff
@@ -160,6 +132,80 @@ impl ScalarBinder {
             ))),
         }
     }
+
+    fn bind_aggregate_op(
+        &self,
+        distinct: bool,
+        func_name: &Identifier,
+        args: &[Expr],
+        params: &[Literal],
+        bind_context: &BindContext,
+    ) -> Result<ScalarExprRef> {
+        let mut data_values = Vec::with_capacity(params.len());
+        for param in params.iter() {
+            data_values.push(match param {
+                Literal::Number(val) => DataValue::try_from_literal(val, None)?,
+                Literal::String(val) => DataValue::String(val.clone().into_bytes()),
+                Literal::Boolean(val) => DataValue::Boolean(*val),
+                Literal::Null => DataValue::Null,
+                Literal::Interval(_) => unimplemented!(),
+                Literal::CurrentTimestamp => unimplemented!(),
+            })
+        }
+
+        let scalar_binder = ScalarBinder::new();
+        let mut scalar_exprs = Vec::with_capacity(args.len());
+        for arg in args.iter() {
+            scalar_exprs.push(
+                scalar_binder
+                    .bind_expr(arg, bind_context)?
+                    .as_any()
+                    .downcast_ref::<Scalar>()
+                    .ok_or_else(|| ErrorCode::UnImplement("Can't downcast to Scalar"))?
+                    .clone(),
+            );
+        }
+
+        let col_pairs = bind_context.result_columns();
+        let mut col_bindings = Vec::with_capacity(col_pairs.len());
+        for col_pair in col_pairs.into_iter() {
+            col_bindings.push(bind_context.resolve_column(None, col_pair.1)?);
+        }
+
+        let mut fields = Vec::with_capacity(col_bindings.len());
+        for col_binding in col_bindings.iter() {
+            fields.push(DataField::new(
+                col_binding.column_name.as_str(),
+                col_binding.data_type.clone(),
+            ))
+        }
+
+        let agg_func_ref = AggregateFunctionFactory::instance().get(
+            func_name.name.clone(),
+            data_values.clone(),
+            fields,
+        )?;
+        let agg_scalar = if optimize_remove_count_args(func_name.name.as_str(), distinct, args) {
+            Scalar::AggregateFunction {
+                func_name: func_name.name.clone(),
+                distinct,
+                params: data_values,
+                args: vec![],
+                data_type: agg_func_ref.return_type()?,
+                nullable: false,
+            }
+        } else {
+            Scalar::AggregateFunction {
+                func_name: func_name.name.clone(),
+                distinct,
+                params: data_values,
+                args: scalar_exprs,
+                data_type: agg_func_ref.return_type()?,
+                nullable: false,
+            }
+        };
```

Comment:
```suggestion
        let agg_scalar = Scalar::AggregateFunction {
            func_name: func_name.name.clone(),
            distinct,
            params: data_values,
            args: if optimize_remove_count_args(func_name.name.as_str(), distinct, args) { vec![] } else { scalar_exprs },
            data_type: agg_func_ref.return_type()?,
            nullable: false,
        };
```

Reply:
Nice, It is surprisingly possible to put `if else` here 😲